torch/distributed/distributed_c10d.py

```diff
@@ -559,6 +564,10 @@ def _new_process_group_helper(world_size,
         raise RuntimeError("Expected timeout argument to be of type"
                            "datetime.timedelta")
 
+    if pg_options is not None and timeout != default_pg_timeout:
```

Comment:
Can we just check if they have the same timeout and continue? Otherwise, if different, raise an error indicating that they are conflicting, and note this in the doc?

Reply:
The check for this condition is actually quite complicated, the current check `if pg_options is not None and timeout != default_pg_timeout and pg_options.timeout != timeout:` mostly works, but if user pass in a `ProcessGroup.Options` with the timeout happen to be `options.timeout == default_pg_timeout`, this will trigger this runtime error, but it's still a legit input. I am wondering if we should just remove this check?